core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-hint-modal.component.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for add hint modal.
+ */
+
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { AppConstants } from 'app.constants';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { StateHintsService } from 'components/state-editor/state-editor-properties-services/state-hints.service';
+import { HintObjectFactory } from 'domain/exploration/HintObjectFactory';
+import { ContextService } from 'services/context.service';
+import { GenerateContentIdService } from 'services/generate-content-id.service';
+import cloneDeep from 'lodash/cloneDeep';
+
+interface HintFormSchema {
+  type: string;
+  'ui_config': object;
+}
+
+@Component({
+  selector: 'oppia-add-hint-modal',
+  templateUrl: './add-hint-modal.component.html'
+})
+export class AddHintModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+  COMPONENT_NAME_HINT: string = AppConstants.COMPONENT_NAME_HINT;
+  tmpHint: string = '';
+  addHintForm = {};
+  hintIndex: number;
+  HINT_FORM_SCHEMA: HintFormSchema = {
+    type: 'html',
+    ui_config: {
+      hide_complex_extensions: (
+        this.contextService.getEntityType() === 'question')
+    }};
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private contextService: ContextService,
+    private generateContentIdService: GenerateContentIdService,
+    private hintObjectFactory: HintObjectFactory,
+    private ngbActiveModal: NgbActiveModal,
+    private stateHintsService: StateHintsService
+  ) {
+    super(ngbActiveModal);
+  }
+
+  ngOnInit(): void {
+    this.hintIndex = this.stateHintsService.displayed.length + 1;
+  }
+
+  getSchema(): HintFormSchema {
```

Comment:
Please add a comment here specifying that `getSchema` function won't be required after migration of `schema-based-editors`.

Reply:
Done.